builder/azure/arm/step_capture_image.go

```diff
@@ -40,58 +40,79 @@ func NewStepCaptureImage(client *AzureClient, ui packersdk.Ui) *StepCaptureImage
 	step.generalizeVM = step.generalize
 	step.captureVhd = step.captureImage
 	step.captureManagedImage = step.captureImageFromVM
-
+	step.getVMInternalID = step.getVMID
 	return step
 }
 
-func (s *StepCaptureImage) generalize(resourceGroupName string, computeName string) error {
-	_, err := s.client.Generalize(context.TODO(), resourceGroupName, computeName)
+func (s *StepCaptureImage) generalize(ctx context.Context, vmId virtualmachines.VirtualMachineId) error {
+	_, err := s.client.Generalize(ctx, vmId)
 	if err != nil {
 		s.say(s.client.LastError.Error())
 	}
 	return err
 }
 
-func (s *StepCaptureImage) captureImageFromVM(ctx context.Context, resourceGroupName string, imageName string, image *compute.Image) error {
-	f, err := s.client.ImagesClient.CreateOrUpdate(ctx, resourceGroupName, imageName, *image)
+func (s *StepCaptureImage) captureImageFromVM(ctx context.Context, subscriptionId string, resourceGroupName string, imageName string, image *images.Image) error {
+	pollingContext, cancel := context.WithTimeout(ctx, s.client.PollingDuration)
+	defer cancel()
+	id := images.NewImageID(subscriptionId, resourceGroupName, imageName)
+	err := s.client.ImagesClient.CreateOrUpdateThenPoll(pollingContext, id, *image)
 	if err != nil {
 		s.say(s.client.LastError.Error())
 	}
-	return f.WaitForCompletionRef(ctx, s.client.ImagesClient.Client)
+	return err
 }
 
-func (s *StepCaptureImage) captureImage(ctx context.Context, resourceGroupName string, computeName string, parameters *compute.VirtualMachineCaptureParameters) error {
-	f, err := s.client.VirtualMachinesClient.Capture(ctx, resourceGroupName, computeName, *parameters)
-	if err != nil {
+func (s *StepCaptureImage) captureImage(ctx context.Context, vmId virtualmachines.VirtualMachineId, parameters *virtualmachines.VirtualMachineCaptureParameters) error {
+	pollingContext, cancel := context.WithTimeout(ctx, s.client.PollingDuration)
+	defer cancel()
+	if err := s.client.VirtualMachinesClient.CaptureThenPoll(pollingContext, vmId, *parameters); err != nil {
 		s.say(s.client.LastError.Error())
+		return err
+	}
+	return nil
+}
+
+func (s *StepCaptureImage) getVMID(ctx context.Context, vmId virtualmachines.VirtualMachineId) (string, error) {
+	vmResponse, err := s.client.VirtualMachinesClient.Get(ctx, vmId, virtualmachines.DefaultGetOperationOptions())
+	if err != nil {
+		return "", err
 	}
-	return f.WaitForCompletionRef(ctx, s.client.VirtualMachinesClient.Client)
+	if vmResponse.Model != nil {
+		vmId := vmResponse.Model.Properties.VMId
+		return *vmId, nil
+	}
+	return "", errors.New(client.SDK_NULL_MODEL_ERROR)
```

Comment:
Instead of creating a constant string to include as the error message. Have you considered creating a custom Error type that can be returned here? 

```go
return "", client.NullModelSDKErr
```

nit: We generally use Camel casing for snake casing for constants and variable names.

Reply:
I originally chose a string cause errors can't be constantly defined, it makes sense imo to just make it a var though so it can be a custom Error, and I've also renamed it with your suggested name in newest commit